### PR TITLE
String Conversion Tests

### PR DESCRIPTION
**Pull request by myty**
Reference: #5

### CODE REVIEW

**Comment by rofr** [comment]
Can't see any files with the commit. Odd!

**Comment by myty** [comment]
Sorry about that.  Should have prefixed WIP to the PR title.  Working on the tests at the moment.

**Comment by myty** [comment]
@rofr  All tests are in now.  Check it out and let me know if I missed anything.

**Comment by rofr** [comment]
So do you think we could use the enum name for enums as well? That would be really nice!

**Comment by myty** [comment]
> So do you think we could use the enum name for enums as well? That would be really nice!

I was thinking about that too.  Seems like it would be a useful feature.  I'll try and whip something up.

**Comment by myty** [comment]
Good news.  Apparently `TypeConverter` already does this.  I added additional tests to handle the name as well.

File: Fig.Core/InvariantStringConverter.cs [hunk 1]
```diff
@@ -13,7 +13,16 @@ public T Convert<T>(string value)
...
         {
             var typeConverter = TypeDescriptor.GetConverter(targetType);
-            return typeConverter.ConvertFromInvariantString(value);
+
+            var convertedValue = typeConverter.ConvertFromInvariantString(value);
+
+            // If target type is an enum, check that the parsed string value is a valid value
+            if (targetType.IsEnum && !Enum.IsDefined(targetType, convertedValue))
+            {
+                throw new ArgumentException($"The string '{nameof(value)}' provided is not a valid Enum value.");
```
Comment (rofr) [inline Fig.Core/InvariantStringConverter.cs]: nameof(value) here will literally yield the string "value".  Probably not what you intended?

Comment (myty) [reply Fig.Core/InvariantStringConverter.cs]: It's what I intended but I can aslo see why that might be confusing.  Will reword to make it little more clear.

Comment (rofr) [reply Fig.Core/InvariantStringConverter.cs]: No worries, I would have included the actual value and also provided a list of valid options. It's useful to have the actual data in the output but obviously could impose a huge security risk. But in any case, there's an ArgumentException constructor overload that takes a message and the argument name.

**Pull request merged**